examples/function_memory_game/src/main.rs

```diff
@@ -0,0 +1,83 @@
+use gloo::timers::callback::{Interval, Timeout};
+use std::{cell::RefCell, rc::Rc};
+use yew::{
+    function_component, html, use_effect_with_deps, use_mut_ref, use_reducer, use_state, Callback,
+};
+
+mod components;
+mod constant;
+mod helper;
+mod state;
+
+use crate::components::{
+    chessboard::Chessboard, game_status_board::GameStatusBoard, score_board::ScoreBoard,
+};
+use crate::constant::Status;
+use crate::state::{Action, State};
+
+#[function_component(App)]
+fn app() -> Html {
```

Comment:
```suggestion
#[function_component]
fn App() -> Html {
```

Reply:
Could you move `App` to its own file `app.rs` ?

Reply:
sure